placeholders/placeholders.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python3
+"""An encoder/decoder for placeholders in python3, using spm vocabulary"""
+import sys
+from typing import List, Dict, Type, Tuple, Optional
+from copy import deepcopy
+from dataclasses import dataclass
+import re
+import argparse
+from random import Random
+from sentencepiece import SentencePieceProcessor
+import yaml
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-c', '--config', type=str, help='Path to yaml configuration file, required for encoding')
+parser.add_argument('-m', '--mappings_file', type=str, default="mappings.yml", help='Path to the mappings, one yaml entry per line.')
+parser.add_argument('-s', '--seed', type=int, default=None, help='Seed for random number generator. Use `-` for sequential numbering.')
```

Comment:
I want to make this optional and the default `1` but I don't understand the comment "Use `-` for sequential mapping. @jelmervdl

Reply:
Ah sorry left over from a little experiment. I've removed that bit again. Right now the default is system random if nothing is specified, and otherwise whatever seed you give it.

Reply:
Oh, didn't know that you could pass None to the constructor... I guess it makes sense.